astropy/utils/data.py

```diff
@@ -167,6 +167,14 @@ def get_readable_fileobj(name_or_obj, encoding=None, cache=False,
     # passed in.  In that case it is not the responsibility of this
     # function to close it: doing so could result in a "double close"
     # and an "invalid file descriptor" exception.
+    PATH_TYPES = six.string_types
+    try:
+        from pathlib import Path
```

Comment:
Better to follow the same idiom from testing and define `HAS_PATHLIB`.  This allows for the possibility of a Python 4 that may or may not have pathlib.

Note: you should change my original implementation in `io.registry` to match (#4405).  I fell into the mistake of testing for `PY3`.  See http://astrofrog.github.io/blog/2016/01/12/stop-writing-python-4-incompatible-code/.  In fact the fix to `io.registry` might not be needed now that you are patching `get_readable_fileobj`.  I haven't looked through the logic that closely but you should look and test.